services/provider/server/server.go

```diff
@@ -1042,3 +1043,143 @@ func (s *OCSProviderServer) PeerStorageCluster(ctx context.Context, req *pb.Peer
 
 	return &pb.PeerStorageClusterResponse{}, nil
 }
+
+func (s *OCSProviderServer) GetStorageClientsInfo(ctx context.Context, req *pb.StorageClientsInfoRequest) (*pb.StorageClientsInfoResponse, error) {
+	var clientsInfo []*pb.ClientInfo
+	var clientError []*pb.StorageClientInfoError
+
+	for i := range req.ClientIDs {
+		consumer, err := s.consumerManager.Get(ctx, req.ClientIDs[i])
+		if err != nil {
+			klog.Errorf("failed to get consumer with id %v: %v", req.ClientIDs[i], err)
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  err.Error(),
+				},
+			)
+			continue
+		}
+		owner := util.FindOwnerRefByKind(consumer, "StorageCluster")
+
+		if owner == nil {
+			klog.Errorf("failed to find StorgeCluster owning the StorageConsumer %v", req.ClientIDs[i])
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "failed to find StorgeCluster owning the StorageConsumer",
+				},
+			)
+			continue
+		}
+
+		if owner.UID != types.UID(req.StorageClusterUID) {
+			klog.Errorf("storageCluster specified on the req does not own the Client")
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_InvalidArgument,
+					Message:  "storageCluster specified on the req does not own the Client",
+				},
+			)
+			continue
+		}
+
+		rnsList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+		err = s.client.List(
+			ctx,
+			rnsList,
+			client.InNamespace(s.namespace),
+			client.MatchingLabels{controllers.StorageConsumerNameLabel: consumer.Name},
+			client.Limit(1),
+		)
+		if err != nil {
+			klog.Error(err)
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "rados namespace for the client not found",
+				},
+			)
+			continue
+		}
+		if len(rnsList.Items) == 0 {
+			klog.Errorf("no rados namespace found for StorageClient %s", req.ClientIDs[i])
+			clientError = append(clientError,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "rados namespace for the client not found: %v",
+				},
+			)
+			continue
+		}
```

Comment:
Same as last comment.

Reply:
If it is not found, since the default value of the string is "", it will always return something. If we don't want to return an error then, we should make radosnamespace field a pointer Have something below this in clientInfo or from the initiator consider "" as an invalid response and requeue
```
message raodsnamespace{
  value = 1;
}
message clientInfo{
  string ClientID
  raodsnamespace raodsnamespace
}
```

Reply:
see reply here: https://github.com/red-hat-storage/ocs-operator/pull/2875#discussion_r1838102551